operators/pkg/instance-controller/logic.go

```diff
@@ -5,18 +5,26 @@ import (
 	"time"
 
 	"github.com/google/uuid"
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	"k8s.io/klog/v2"
-	ctrl "sigs.k8s.io/controller-runtime"
 
 	crownlabsv1alpha2 "github.com/netgroup-polito/CrownLabs/operators/api/v1alpha2"
 	instance_creation "github.com/netgroup-polito/CrownLabs/operators/pkg/instance-creation"
+
+	"github.com/prometheus/common/log"
+
+	"k8s.io/apimachinery/pkg/types"
+
+	"k8s.io/klog"
+	v1 "kubevirt.io/client-go/api/v1"
+	virtv1 "kubevirt.io/client-go/api/v1"
+	cdiv1 "kubevirt.io/containerized-data-importer/pkg/apis/core/v1alpha1"
+	ctrl "sigs.k8s.io/controller-runtime"
 )
 
 func (r *LabInstanceReconciler) CreateVMEnvironment(labInstance *crownlabsv1alpha2.Instance, environment *crownlabsv1alpha2.Environment, namespace, name string, vmStart time.Time) error {
 	var user, password string
+	var vmi *v1.VirtualMachineInstance
 	// this is added so that all resources created for this Instance are destroyed when the Instance is deleted
-	b := true
+	/*b := true
```

Comment:
Some questions:
1. Commented code is dead code. If we have to delete it, let's do it (Git will store the change for us).
2. Why are we deleting such ownerReference?

Reply:
We have deleted commented code.
We are deleting GlobalOwnerReference in order to pass to the function SetControllerReference ( function that belong to the controller-runtime package). Moreover we have changed the owner references of ingresses, services and secrets, that now belong to VMIs. In this way, in case of non persistent VMIs nothing changes (because VMIs are still owned by instances). However in the case of persistent VMIs, when this is deleted you should not need to have secrets, ingress and services running so also those are deleted. Whenever you will create again the VMis those resources are recreated